drivers/i2c/i2c_ll_stm32.c

```diff
@@ -127,14 +143,33 @@ i2c_stm32_transaction(const struct device *dev,
 			msg.flags = saved_flags;
 			flagsp = next_msg_flags;
 		}
+
+#ifdef CONFIG_PM
+		/* Prevent SoC from entering power states defined in
+		 * reinit-power-states DT property
+		 */
+		DT_FOREACH_PROP_ELEM_VARGS(
+			DT_DRV_INST(0), reinit_power_states,
+			PM_POLICY_STATE_LOCK_ACTION, pm_policy_state_lock_get);
+#endif
```

Comment:
Can we think of a generic, dedicated macro ? (ideally with CONFIG_PM=n abstraction)
Also, it should take an argument that would be the "lowest level unless specified otherwise in dt" so that `reinit-power-states` can remain optional.

Reply:
I'll attempt to create an intermediate macro for generating this logic directly.

Reply:
Please see https://github.com/zephyrproject-rtos/zephyr/pull/60369#issuecomment-1641248231